cpp/src/binaryop/compiled/binops_custom.cuh

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "binary_ops.hpp"
+#include "operation.cuh"
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+
+namespace binops {
+namespace compiled {
+
+namespace {
+
+// Functors to launch only defined operations.
+
+/**
+ * @brief Functor to launch only defined operations with common type.
+ *
+ * @tparam BinaryOperator binary operator functor
+ */
+template <typename BinaryOperator, bool store_as_result = false>
+struct ops_wrapper {
+  mutable_column_device_view& out;
+  column_device_view const& lhs;
+  column_device_view const& rhs;
+  template <typename TypeCommon>
+  __device__ void operator()(size_type i)
```

Comment:
For operators such as this that have all usages of `i` instead a `if constexpr` block we will need to add `(void) i;` to the end of the method. The compiler when compiling for the false side of the constexpr branch will see no usages of `i` and generate a warning that we have a unused variable ( nvcc 11.3+ ).